junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/JupiterTestDescriptor.java

```diff
@@ -88,9 +99,22 @@
 		// @formatter:on
 	}
 
-	protected static <E extends AnnotatedElement> String determineDisplayName(E element,
-			Function<E, String> defaultDisplayNameGenerator) {
+	protected static DisplayNameGenerator getDisplayNameGenerator(Class<?> testClass) {
+		Preconditions.notNull(testClass, "Test class must not be null");
+		Optional<ExtendWith> optionalExtendWith = findAnnotation(testClass, ExtendWith.class);
+		if (optionalExtendWith.isPresent()) {
+			for (Class<?> candidate : optionalExtendWith.get().value()) {
+				if (!DisplayNameGenerator.class.isAssignableFrom(candidate)) {
+					continue;
+				}
+				return (DisplayNameGenerator) ReflectionUtils.newInstance(candidate);
```

Comment:
This is _quick and dirty_ hack to get a handle to a user implementation of the `DisplayNameGenerator` interface.

Having support for such a declarative way is debatable ... and if it is needed/wanted, it shouldn't use the `ExtendWith` annotation.

Reply:
I agree: let's get rid of the hack and do something better suited to the task at hand.

I'll add some notes to the original issue rather than discuss that here in this PR.